### PR TITLE
[NL] Add a toolformer mode

### PR DESCRIPTION
**Pull request by chejennifer**
Adds a new toolformer mode that:
- skips topics when detecting SVs
- has a sv score threshold of 0.8

this new mode can be used in both dc explore and node server by adding the param `&mode=toolformer`

example: what is the infant mortality rate in massachusetts
regular explore: https://screenshot.googleplex.com/C52Wep5C5kWcFFh
toolformer mode: https://screenshot.googleplex.com/4j9wcNAHpbFTavN

### CODE REVIEW

File: nl_server/embeddings.py [hunk 1]
```diff
@@ -35,6 +35,11 @@
 
 # Number of matches to find within the SV index.
 _NUM_SV_INDEX_MATCHES = 40
+# Number of matches to find within the SV index if skipping topics.
+_SKIP_TOPICS_NUM_SV_INDEX_MATCHES = 60
```
Comment (pradh) [inline nl_server/embeddings.py:39]: super nit:  _NUM_SV_INDEX_MATCHES_WITHOUT_TOPICS

File: server/lib/explore/params.py [hunk 2]
```diff
@@ -51,6 +51,7 @@ class DCNames(str, Enum):
 class QueryMode(str, Enum):
   # NOTE: This mode is incompatible with LLM detector
   STRICT = 'strict'
+  TOOLFORMER = 'toolformer'
```
Comment (pradh) [inline server/lib/explore/params.py:54]: Comment what this is?

File: server/lib/explore/params.py [hunk 3]
```diff
@@ -61,6 +62,8 @@ class Clients(str, Enum):
 def sv_threshold(mode: str) -> bool:
   if mode == QueryMode.STRICT:
     return constants.SV_SCORE_HIGH_CONFIDENCE_THRESHOLD
+  elif mode == mode == QueryMode.TOOLFORMER:
```
Comment (pradh) [inline server/lib/explore/params.py:65]: two `mode ==` :)

File: static/src/server.ts [hunk 4]
```diff
@@ -503,12 +505,13 @@ app.get("/nodejs/query", (req: Request, res: Response) => {
...
   const allowedTileTypes =
     client === BARD_CLIENT_URL_PARAM ? BARD_ALLOWED_CHARTS : null;
+  const mode = req.query.mode || DEFAULT_NL_MODE;
   res.setHeader("Content-Type", "application/json");
   axios
     // Set "mode=strict" to use heuristic detector, disable using default place,
     // use a higher SV threshold and avoid multi-verb queries
     .post(
-      `${CONFIG.apiRoot}/api/explore/detect-and-fulfill?q=${query}&mode=strict&client=${client}`,
+      `${CONFIG.apiRoot}/api/explore/detect-and-fulfill?q=${query}&mode=${mode}&client=${client}`,
```
Comment (pradh) [inline static/src/server.ts:514]: Maybe follow up, could we maybe add SV/place DCIDs in the charts?  Without that now its hard to know which chart has what SVs.

**Review by pradh — APPROVED** [review]
Nice!

File: nl_server/embeddings.py (hunk 1 above)
Comment (chejennifer) [reply nl_server/embeddings.py]: done

File: server/lib/explore/params.py (hunk 2 above)
Comment (chejennifer) [reply server/lib/explore/params.py:56]: done

File: server/lib/explore/params.py (hunk 3 above)
Comment (chejennifer) [reply server/lib/explore/params.py]: good catch, fixed

File: static/src/server.ts (hunk 4 above)
Comment (chejennifer) [reply static/src/server.ts:514]: do you mean in the json response for each chart include additional fields for SV and place DCIDs?

Comment (pradh) [reply static/src/server.ts:514]: Yes.

**Review by pradh — APPROVED** [review]

**Comment by chejennifer** [comment]
thanks for the review!

**Pull request merged**